packages/react/src/components/Text/TextDirection.tsx

```diff
@@ -9,7 +9,7 @@ import PropTypes from 'prop-types';
 import React, { ReactNode, useEffect, useMemo, useRef } from 'react';
 import { TextDirectionContext } from './TextDirectionContext';
 
-export type TextDir = 'ltr' | 'rtl' | 'auto';
+export type TextDir = 'ltr' | 'rtl' | 'auto' | string;
```

Comment:
Needed to add `string` because some components that were changed to `Text` were spreading `...other`. This was causing TypeScript errors because the types for `dir` are `string | undefined` on `ComponentProps` (`ListItem` was throwing this)